rule/add_constant.go

```diff
@@ -40,10 +41,12 @@ type AddConstantRule struct {
 }
 
 // Apply applies the rule to given file.
-func (r *AddConstantRule) Apply(file *lint.File, arguments lint.Arguments) []lint.Failure {
-	r.configure(arguments)
-
+func (r *AddConstantRule) Apply(file *lint.File, arguments lint.Arguments) ([]lint.Failure, error) {
 	var failures []lint.Failure
+	err := r.configure(arguments)
```

Comment:
A recently merged PR (merged after your fork) introduced the use of `sync.Once` as a replacement for `sync.Mutex` in configuration functions. Even nice per se, the use of `sync.Once` complicates the retrieval of configuration results (i.e. retrieving the error result is not as straightforward as `err := r.configure(arguments)`)

Reply:
@chavacava any suggestion how to do that in proper way?

Reply:
@chavacava aaah should I merge changes from https://github.com/mgechev/revive/pull/1118/files, before made changes with ```err := r.configure(arguments)```

Reply:
ok @chavacava I added to each rule struct additional field ```configureErr    error``` and then update:

```r.configureOnce.Do(func() { r.configure(arguments) })```

to:

```
r.configureOnce.Do(func() { r.configureErr = r.configure(arguments) })                                                                                                 
                                                                                                                                                                            
if r.configureErr != nil {                                                                                                                                             
  return nil, r.configureErr                                                                                                                                         
}
```  
after that with some changes in config file (make some invalid arguments), revive stops and prints on stdout correct error (of course related to rule)